pasqal_cloud/errors.py

```diff
@@ -8,7 +8,12 @@ class ExceptionWithResponseContext(BaseException):
     def __init__(self, msg: str, e: Optional[HTTPError] = None) -> None:
         if not e:
             return super().__init__(msg)
+
+        if e.response is None:
```

Comment:
There is a mock library used in the SDK tests that when returning an exception instead of a payload, does not return response objects, so this change was introduced, which shouldn't provide issues with standard use